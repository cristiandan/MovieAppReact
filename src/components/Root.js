import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'react-redux';
import App from './App';
import { BrowserRouter, Route, Switch } from 'react-router-dom'

import HomePage from '../containers/HomePageContainer';
import NotFoundPage from '../components/NotFoundPage';
import SearchPage from '../containers/SearchPageContainer';
import MovieDetails from '../containers/MovieDetailsContainer';

export default class Root extends Component {
  render() {
    const { store } = this.props;
    return (
      <Provider store={store}>
        <div>
          <BrowserRouter>
            <div>
              <App />
              <Switch>
                  <Route exact path="/" component={HomePage}/>
                  <Route exact path="/search" component={SearchPage}/>
                  <Route exact path="/movie/:id" component={MovieDetails} />
                  <Route component={NotFoundPage}/>
              </Switch>
            </div>
          </BrowserRouter>
        </div>
      </Provider>
    );
  }
}

Root.propTypes = {
  store: PropTypes.object.isRequired,
};