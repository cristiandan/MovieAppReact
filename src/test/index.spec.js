import { mount } from 'enzyme';
import React from 'react';
import toJson from 'enzyme-to-json';
import Root from '../components/Root';
import configureStore from '../store/configureStore';

const store = configureStore();

describe('The app can', async () => {
    it('render', async () => {
        const app = mount(<Root store={store} />);
        await new Promise(r => setTimeout(r, 0));
        app.update();
        app.find('a[id="homelink"]').simulate('click')
        expect(toJson(app.render())).toMatchSnapshot();
    })
})