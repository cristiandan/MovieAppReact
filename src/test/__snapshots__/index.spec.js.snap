// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The app can render 1`] = `
<div>
  <div>
    <div>
      <a
        href="/"
        id="homelink"
      >
        Home
      </a>
       | 
      <a
        href="/search"
        id="searchlink"
      >
        Search
      </a>
       | 
      <a
        href="/about"
        id="aboutlink"
      >
        About
      </a>
      <br>
        
      </br>
    </div>
    <div>
      <h4>
        404 Page Not Found
      </h4>
      <a
        href="/"
      >
         Go back to homepage 
      </a>
    </div>
  </div>
</div>
`;
